language/vm/src/file_format.rs

```diff
@@ -338,8 +338,37 @@ pub struct FieldInstantiation {
     pub type_parameters: SignatureIndex,
 }
 
-/// A `StructDefinition` is a type definition. It either indicates it is native or
-// defines all the user-specified fields declared on the type.
+/// `Visibility` restricts the accessibility of the associated entity.
+/// - For function visibility, it restricts who may call into the associated function.
+#[derive(Clone, Debug, Eq, PartialEq)]
+#[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary))]
+#[cfg_attr(any(test, feature = "fuzzing"), proptest(no_params))]
+pub enum Visibility {
+    Private,
+    Protected,
+    Public,
+}
+
+impl Default for Visibility {
+    fn default() -> Self {
+        Visibility::Private
+    }
+}
+
+/// A `FriendDeclaration` encapsulates information about a friend.
+#[derive(Clone, Debug, Hash, Eq, PartialEq)]
+#[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary))]
+#[cfg_attr(any(test, feature = "fuzzing"), proptest(no_params))]
+pub struct FriendDeclaration {
+    /// Handle index to the friend module
+    pub module: ModuleHandleIndex,
+    // TODO: for future extensions, we may want to attach policies to friend declarations, e.g.,
+    // in the case of cross-address friendship, we may have a policy controlling whether the callee
+    // needs to sign if the caller wants to make an update.
+}
```

Comment:
We can probably remove this struct?

Reply:
I'm guessing you added this because of the TODO of more policies, but there isn't really any point to having this struct now. And we will have to bump the fileformat version and do a refactor for any extension 

So best to not have the layer of indirection here if we don't need it

Reply:
Yes, the main reason for a struct here is for the ease of extensibility. But yes, the struct is not really needed now.

Reply:
Yeah I don't see it making it easier to making things extensible in the future. As in, the amount of work to actually extend anything isn't reduced by this struct

Reply:
After some refactoring trials I still want to keep the struct even if there is only one field here. It kinds of gives me a mentality that I am dealing with a "friend" instead of just a ModuleId, and we just happen to define that the unit of "friend" is a module.

But I also see the point that it may be good to ease the concept of "friend" out from the bytecode level, especially since we haven't fully decided on the name, so the fewer "friend" in var/struct names, the better.

Tough choice.